File: darkroom-web/src/services/mock/release.ts
import type { ReleaseInfo } from "@/types";
import { mockSongs } from "./songs";
import { getSongMeta } from "./songs";

/** MOCK release records. No release backend exists yet. */
const STAGES: ReleaseInfo["stage"][] = [
  "Preparing",
  "Ready",
  "Scheduled",
  "Released",
  "Planning",
  "Scheduled",
  "Planning",
  "Planning",
];

const CHECKLIST_DONE: number[] = [3, 5, 6, 7, 0, 5, 0, 1];

const CHECKLIST_ITEMS = [
  "Master",
  "Cover",
  "Metadata",
  "Credits",
  "Canvas",
  "Music Video",
  "Content Plan",
];

export function getRelease(songId: string): ReleaseInfo {
  const index = mockSongs.findIndex((s) => s.id === songId);
  const meta = getSongMeta(songId);
  const doneCount = index >= 0 ? CHECKLIST_DONE[index] ?? 0 : 0;

  return {
    songId,
    releaseDate: meta.releaseDate ?? "Not scheduled",
    distributor: index >= 0 ? "DISTROKID" : "—",
    isrc: index >= 0 ? `QZ-K4S-26-00${index + 1}` : "—",
    upc: index >= 0 ? `19122${(700000 + index * 37).toString()}` : "—",
    releaseType: "Single",
    platforms: ["Spotify", "Apple Music", "YouTube Music", "Tidal", "Deezer"],
    stage: (index >= 0 ? STAGES[index] : "Planning") ?? "Planning",
    checklist: CHECKLIST_ITEMS.map((item, i) => ({ item, done: i < doneCount })),
  };
}

File: ArtistOS.Api/Models/Release.cs
using System.ComponentModel.DataAnnotations;

namespace ArtistOS.Api.Models;

public class Release
{
    public int Id { get; set; }

    public int SongId { get; set; }

    public Song Song { get; set; } = null!;

    public DateOnly? ReleaseDate { get; set; }

    [MaxLength(40)]
    public string ReleaseType { get; set; } = "Single";

    [MaxLength(120)]
    public string? Distributor { get; set; }

    [MaxLength(20)]
    public string? Isrc { get; set; }

    [MaxLength(20)]
    public string? Upc { get; set; }

    [MaxLength(40)]
    public string Status { get; set; } = "Planning";

    [MaxLength(255)]
    public string Platforms { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}
